langgraph/src/pregel/index.ts

```diff
@@ -326,16 +341,25 @@ export class Pregel
 
     _applyWrites(checkpoint, channels, inputPendingWrites);
 
-    const read = (chan: string) => readChannel(channels, chan);
-
     // Similarly to Bulk Synchronous Parallel / Pregel model
     // computation proceeds in steps, while there are channel updates
     // channel updates from step N are only visible in step N+1
     // channels are guaranteed to be immutable for the duration of the step,
     // with channel updates applied only at the transition between steps
     const recursionLimit = config.recursionLimit ?? DEFAULT_RECURSION_LIMIT;
     for (let step = 0; step < recursionLimit + 1; step += 1) {
-      const nextTasks = _prepareNextTasks(checkpoint, processes, channels);
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const [nextCheckpoint, nextTasks] = _prepareNextTasks(
+        checkpoint,
+        processes,
+        channels,
+        true
+      );
+
+      // Reassign nextCheckpoint to checkpoint because the subsequent implementation
+      // relies on side effects applied to checkpoint. Example: _applyWrites().
+      checkpoint = nextCheckpoint as Checkpoint;
```

Comment:
This is temporary. When `stream()` is reimplemented, I think this code will be removed.